android/src/main/java/com/pspdfkit/react/helper/DocumentJsonDataProvider.java

```diff
@@ -0,0 +1,50 @@
+package com.pspdfkit.react.helper;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.pspdfkit.document.providers.DataProvider;
+
+import org.json.JSONObject;
+
+import java.nio.charset.Charset;
+
+/**
+ * {@link DataProvider} that is used when adding document instant json.
+ */
+public class DocumentJsonDataProvider implements DataProvider {
```

Comment:
Huh, do we internally require a `DataProvider` for parsing JSON? Seems a bit verbose to me, and we probably don't require random access to JSON data. We should discuss this with the team.

Reply:
Yes we do, see Documentation here https://pspdfkit.com/guides/android/current/importing-exporting/instant-json/#applying-document-json-to-a-document

Reply:
Seems like an arbitrary requirement to me, something we should reevaluate.